test/converters/xcresult/xcresult.go

```diff
@@ -0,0 +1,188 @@
+package xcresult
+
+import (
+	"encoding/xml"
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"howett.net/plist"
+
+	"github.com/bitrise-io/go-utils/fileutil"
+)
+
+//
+// TestSummaries.plist
+
+type TestSummaryPlist struct {
+	FormatVersion     string
+	TestableSummaries []TestableSummary
+}
+
+func collapseSubtestTree(data Subtests) (tests Subtests) {
+	for _, test := range data {
+		if len(test.Subtests) > 0 {
+			tests = append(tests, collapseSubtestTree(test.Subtests)...)
+		}
+		if test.TestStatus != "" {
+			tests = append(tests, test)
+		}
+	}
+	return
+}
+
+func (summaryPlist TestSummaryPlist) Tests() map[string]Subtests {
+	tests := map[string]Subtests{}
+	var subTests Subtests
+	for _, testableSummary := range summaryPlist.TestableSummaries {
+		for _, test := range testableSummary.Tests {
+			subTests = append(subTests, collapseSubtestTree(test.Subtests)...)
+		}
+	}
+	for _, test := range subTests {
+		testID := strings.Split(test.TestIdentifier, "/")[0]
+		tests[testID] = append(tests[testID], test)
+	}
+	return tests
+}
+
+type TestableSummary struct {
+	TargetName      string
+	TestKind        string
+	TestName        string
+	TestObjectClass string
+	Tests           []Test
+}
+
+type FailureSummary struct {
+	FileName           string
+	LineNumber         int
+	Message            string
+	PerformanceFailure bool
+}
+
+type Subtest struct {
+	Duration         float64
+	TestStatus       string
+	TestIdentifier   string
+	TestName         string
+	TestObjectClass  string
+	Subtests         Subtests
+	FailureSummaries []FailureSummary
+}
+
+func (st Subtest) Failure() (message string) {
+	prefix := ""
+	for _, failure := range st.FailureSummaries {
+		message += fmt.Sprintf("%s%s:%d - %s", prefix, failure.FileName, failure.LineNumber, failure.Message)
+		prefix = "\n"
+	}
+	return
+}
+
+type Subtests []Subtest
+
+func (sts Subtests) FailuresCount() (count int) {
+	for _, test := range sts {
+		if len(test.FailureSummaries) > 0 {
+			count++
+		}
+	}
+	return count
+}
+
+func (sts Subtests) TotalTime() (time float64) {
+	for _, test := range sts {
+		time += test.Duration
+	}
+	return time
+}
+
+type Test struct {
+	Subtests Subtests
```

Comment:
Not sure about this one

Reply:
It is because the xml unmarshal, this is the only way I could group xml elements as the standard junit4 xml.

Reply:
This is how I could follow the relevant plist structure.